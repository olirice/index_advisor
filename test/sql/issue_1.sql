begin;

    create extension index_advisor version '0.1.1' cascade;

    create table public.book(
        id int,
        -- json type is not btree indexable. In version 0.1.1 this raises the error
        -- ERROR:  data type json has no default operator class for access method "btree"
        meta json
    );

    select index_advisor($$
        select * from book where id = $1
    $$);

rollback;
