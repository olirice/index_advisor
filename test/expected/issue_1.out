begin;
    create extension index_advisor version '0.2.0' cascade;
NOTICE:  installing required extension "hypopg"
    create table public.book(
        id int,
        -- json type is not btree indexable. In version 0.1.1 this raises the error
        -- ERROR:  data type json has no default operator class for access method "btree"
        meta json
    );
    select index_advisor($$
        select * from book where id = $1
    $$);
                                  index_advisor                                  
---------------------------------------------------------------------------------
 (0.00,4.07,25.88,13.54,"{""CREATE INDEX ON public.book USING btree (id)""}",{})
(1 row)

rollback;
